Bottom: dd6067b005c74806c5d90f5f920557b765e5404e
Top:    8847eb5f69a68a80926469ff64a2d1e0c112e729
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2015-03-23 00:13:07 +0000

PCB+GL: Fix GL headers for Win32


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 732cef5..01a4479 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -32,12 +32,18 @@
 #include <math.h>
 #include <assert.h>
 
-/* The Linux OpenGL ABI 1.0 spec requires that we define
- * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
- * in order to get prototypes:
- *   http://www.opengl.org/registry/ABI/
- */
-#define GL_GLEXT_PROTOTYPES 1
+#ifdef WIN32
+#   define WIN32_LEAN_AND_MEAN 1
+#   include <windows.h>
+#else
+    /* The Linux OpenGL ABI 1.0 spec requires that we define
+     * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
+     * in order to get prototypes:
+     *   http://www.opengl.org/registry/ABI/
+     */
+#   define GL_GLEXT_PROTOTYPES 1
+#endif
+
 #ifdef HAVE_OPENGL_GL_H
 #   include <OpenGL/gl.h>
 #else
